apis/apps/v1alpha1/apicast_types.go

```diff
@@ -181,6 +192,26 @@ func (a *APIcast) GetOwnerRefence() *metav1.OwnerReference {
 
 func (a *APIcast) Reset() { *a = APIcast{} }
 
+func (a *APIcast) Validate() field.ErrorList {
+	errors := field.ErrorList{}
+
+	// check HTTPSPort is required for httpsCertificateSecretRef
```

Comment:
Should using a default port be a good idea?

Reply:
What does trigger enablement of HTTPs in APIcast? specifying the port? or the certifcate?

Reply:
I guess this is a decision up to us. The APIcast will not activate https unless HTTPSPort is provided in env var.

The current implementation ensures both (certificate and https port) are provided or none. Another approach is that if the customer provides the certificate and does not provide the port, the operator takes a default one on behalf of the customer (for instance 443 or 8081). I think this approach requires less config from the customer. I will implement.

Reply:
implemented default port approach. Ready for review.

Reply:
:+1: